functional-tests/src/main/java/com/redhat/che/selenium/core/client/RhCheTestWorkspaceServiceClient.java

```diff
@@ -87,17 +97,41 @@ public Workspace createWorkspace(
   @Override
   public void start(String workspaceId, String workspaceName, TestUser workspaceOwner)
       throws Exception {
+    if (getStatus(workspaceId).equals(WorkspaceStatus.RUNNING)) {
+      LOG.info("Workspace is running - no need to start it.");
+      return;
+    }
     try {
       this.cheStarterWrapper.startWorkspace(workspaceId, workspaceName, token);
       waitStatus(workspaceName, owner.getName(), WorkspaceStatus.RUNNING);
       LOG.info("Workspace " + workspaceName + "is running.");
     } catch (Exception e) {
-
       LOG.error("Failed to start workspace \"" + workspaceName + "\": " + e.getMessage(), e);
       throw e;
     }
   }
 
+  @Override
+  public void waitStatus(String workspaceName, String userName, WorkspaceStatus expectedStatus)
+      throws Exception {
+    WorkspaceStatus status = null;
+    for (int i = 0; i < 600; i++) {
+      status = getByName(workspaceName, userName).getStatus();
+      if (status == expectedStatus) {
+        return;
+      } else {
+        if (expectedStatus == WorkspaceStatus.RUNNING && status == WorkspaceStatus.STOPPED) {
```

Comment:
This `if` is the only difference between this method and the one from parent, right? What's the purpose of this?

Reply:
Yes, this is the only difference. The reason is that the default timeout for starting workspace is set by upstream to 10 mins. As far as I know there is not any workflow when workspace can became "STOPPED" when starting. This ```if``` allows it to be detected much earlier, so time of tests will be shorter and it will be more friendly when testing locally. Next reason why I decided to add it is that we added it in our old tests too.
On the other hand, this behaviour is not needed, so if you prefer, I can remove it.

Reply:
I would prefer removing this re-implementation. If we find ourself in situation, where we need that, I would prefer change upstream (adding parameter, where the timeout can be changed, or changing the logic there...)

Reply:
@rhopp agreed.

Reply:
:+1: